homeassistant/helpers/config_validation.py

```diff
@@ -89,14 +89,45 @@ def validate(obj: Dict) -> Dict:
 
 def boolean(value: Any) -> bool:
     """Validate and coerce a boolean value."""
+    if value is None:
+        return False
+    if isinstance(value, bool):
+        return value
     if isinstance(value, str):
-        value = value.lower()
+        value = value.lower().strip()
         if value in ('1', 'true', 'yes', 'on', 'enable'):
             return True
         if value in ('0', 'false', 'no', 'off', 'disable'):
             return False
         raise vol.Invalid('invalid boolean value {}'.format(value))
-    return bool(value)
+    if isinstance(value, Number):
+        return value != 0
+    raise vol.Invalid('invalid boolean value {}'.format(value))
+
+
+def boolean_tolerant(value: Any) -> bool:
```

Comment:
Can't you just define this one as 

```python
boolean_tolerant = vol.Any(boolean, lambda val: False)
```

That way we're not duplicating logic.

I actually think that maybe we should call it `coerce_boolean`, because it's always returning a boolean?

Reply:
Yep that's sensible. The reason I wanted them next to each other in the source is so the concept of true was uniform and didn't drift in different locations, that's even better.